build_metal.sh

```diff
@@ -0,0 +1,44 @@
+
+if [ -z "$PYTHON_ENV_DIR" ]; then
+    PYTHON_ENV_DIR=$(pwd)/python_env
+fi
+python3 -m venv $PYTHON_ENV_DIR
+
+source $PYTHON_ENV_DIR/bin/activate
+
+echo "Setting up virtual env"
+python3 -m pip config set global.extra-index-url https://download.pytorch.org/whl/cpu
+python3 -m pip install setuptools wheel
+
+echo "Installing dev dependencies"
+python3 -m pip install -r $(pwd)/tt_metal/python_env/requirements-dev.txt
+
+if [ -z "$CONFIG" ]; then
+    echo "Build type defaulted to RelWithDebInfo (assert)"
+else
+    VALID_CONFIGS="RelWithDebInfo Debug ci"
+    if [[ $VALID_CONFIGS =~ (^|[[:space:]])"$CONFIG"($|[[:space:]]) ]]; then
+        echo "CONFIG set to $CONFIG"
+    else
+        echo "Invalid config "$CONFIG" given.. Valid configs are: $VALID_CONFIGS"
+        exit 1
+    fi
+fi
+
+echo "Building/installing tt-metal"
+pip install -e .
+pip install -e ttnn
```

Comment:
does it make sense to create a separate `ttnn-dev` wheel that brings in dependencies found in `requirements-dev.txt` automatically? @tt-rkim